src/Common/tests/System/PlatformDetection.cs

```diff
@@ -278,26 +319,77 @@ private static IdVersionPair ParseOsReleaseFile()
                 {
                     if (line.StartsWith("ID=", System.StringComparison.Ordinal))
                     {
-                        ret.Id = line.Substring("ID=".Length);
+                        ret.Id = RemoveQuotes(line.Substring("ID=".Length));
                     }
                     else if (line.StartsWith("VERSION_ID=", System.StringComparison.Ordinal))
                     {
-                        ret.VersionId = line.Substring("VERSION_ID=".Length);
+                        ret.VersionId = RemoveQuotes(line.Substring("VERSION_ID=".Length));
                     }
                 }
             }
-
-            string versionId = ret.VersionId;
-            if (versionId.Length >= 2 && versionId[0] == '"' && versionId[versionId.Length - 1] == '"')
+            else 
             {
-                // Remove quotes.
-                ret.VersionId = versionId.Substring(1, versionId.Length - 2);
-            }
+                string fileName = null;
+                if (File.Exists("/etc/redhat-release"))
```

Comment:
Would it be beneficial to use the [`RuntimeEnvironment`](https://github.com/dotnet/core-setup/blob/9dff3ce12574699e6d4d31722047f775d92d05fd/src/managed/Microsoft.DotNet.PlatformAbstractions/RuntimeEnvironment.cs#L9) class here?  It is already doing all this logic, and it is how our customers get the current RID of the machine, and what version of RedHat we are on, etc.

I know the dependency would be a little inverted here (corefx code referencing core-setup code).  But we do that today when using the "shared framework" to run tests.  So there is precedent.

/cc @weshaggard

Reply:
That would make core-setup necessary for running dotnet apps. Now it is not needed - you can just build coreclr and corefx and use corerun  / coreconsole for running your app. I believe that's what the Samsung folks are using. 
If we want to unify the detection code (which would be great), I think we should rather move it to corefx and let core-setup use it.

Reply:
I thought this was only for test code?

As for moving the real logic into CoreFX- yes please!

Reply:
Ah, I am sorry, I've missed that it is for tests only. You are right.

Reply:
@eerhardt does RuntimeEnvironment is a public class which can easily take a dependency on? Also, how easy we can add more properties to it? I am asking because for the tests we need to be flexible of adding more properties to the PaltformDetection code and customize it to fit the cases needed in the tests. in another word, for tests, I prefer the flexibility even if we duplicate some of such features. 

Another thing, this PR is for 2.0 servicing branch which I prefer to minimize the changes here, so if we decide to take a dependency on RuntimeEnvironment, I prefer to do it in master and not in 2.0 branch.

Reply:
Yes we should be able to take a dependency on it. We shouldn't be adding a bunch test specific properties to that library though. We should still keep our PlatformDetection code but instead of duplicating all the platform detection we should call into the RuntimeEnvironment class instead.

Reply:
@weshaggard why this functionality OS not exposed from RuntimeInformation class instead? why have we introduced RuntimeEnvironment while this can fit better in RuntimeInformation?

Reply:
It may get moved down eventually but today the RID stuff is only on the RuntimeEnviroment class.

Reply:
While I see advantage of using RuntimeEnvironment here I'll exclude this change request from 2.0.0 PR as it will introduce too much churn because PlatformDetection currently lives in the Common folder meaning every test project will now have to depend on PlatformAbstraction. I'll change it in the PR against master as we can change just test utils project.

If you strongly feel we should do that in 2.0.0 branch I'll do it but currently think we should limit 2.0.0 changes minimum.

Reply:
To clarify, I'll do this for master branch but will exclude for servicing branch (because too many projects will have to be touched)

Reply:
> Another thing, this PR is for 2.0 servicing branch which I prefer to minimize the changes here, so if we decide to take a dependency on RuntimeEnvironment, I prefer to do it in master and not in 2.0 branch.

> I'll do this for master branch but will exclude for servicing branch (because too many projects will have to be touched)

Works for me.  Thanks.